<a href="https://colab.research.google.com/github/eboekenh/Agentic-AI-Bootcamp/blob/main/Answers-First-Assignmen/4_Answers_Pandas_Exercise_2_Movie_Dataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Practice Exercise 2

In this assignment, you will try to find some interesting insights into a few movies released between 1916 and 2016, using Python. You will have to download a movie dataset, write Python code to explore the data, gain insights into the movies, actors, directors, and collections, and submit the code.

#### Some tips before starting the assignment

1. Identify the task to be performed correctly, and only then proceed to write the required code. Don’t perform any incorrect analysis or look for information that isn’t required for the assignment.
2. In some cases, the variable names have already been assigned, and you just need to write code against them. In other cases, the names to be given are mentioned in the instructions. We strongly advise you to use the mentioned names only.
3. Always keep inspecting your data frame after you have performed a particular set of operations.
4. There are some checkpoints given in the IPython notebook provided. They're just useful pieces of information you can use to check if the result you have obtained after performing a particular task is correct or not.
5. Note that you will be asked to refer to documentation for solving some of the questions. That is done on purpose for you to learn new commands and also how to use the documentation.

In [141]:
# Import the numpy and pandas packages

import numpy as np
import pandas as pd

### Task 1: Reading and Inspection

**Subtask 1.1: Import and read**

Import and read the movie database. Store it in a variable called `movies`.

In [142]:
# Write your code for importing the csv file here
movies = pd.read_csv('4_Movies.csv')
movies

,color,director_name,num_critic_for_reviews,duration,director_facebook_likes,actor_3_facebook_likes,actor_2_name,actor_1_facebook_likes,gross,genres,...,num_user_for_reviews,language,country,content_rating,budget,title_year,actor_2_facebook_likes,imdb_score,aspect_ratio,movie_facebook_likes
0,Color,James Cameron,723.0,178.0,0.0,855.0,Joel David Moore,1000.0,760505847.0,Action|Adventure|Fantasy|Sci-Fi,...,3054.0,English,USA,PG-13,237000000.0,2009.0,936.0,7.9,1.78,33000
1,Color,Gore Verbinski,302.0,169.0,563.0,1000.0,Orlando Bloom,40000.0,309404152.0,Action|Adventure|Fantasy,...,1238.0,English,USA,PG-13,300000000.0,2007.0,5000.0,7.1,2.35,0
2,Color,Sam Mendes,602.0,148.0,0.0,161.0,Rory Kinnear,11000.0,200074175.0,Action|Adventure|Thriller,...,994.0,English,UK,PG-13,245000000.0,2015.0,393.0,6.8,2.35,85000
3,Color,Christopher Nolan,813.0,164.0,22000.0,23000.0,Christian Bale,27000.0,448130642.0,Action|Thriller,...,2701.0,English,USA,PG-13,250000000.0,2012.0,23000.0,8.5,2.35,164000
4,Color,Andrew Stanton,462.0,132.0,475.0,530.0,Samantha Morton,640.0,73058679.0,Action|Adventure|Sci-Fi,...,738.0,English,USA,PG-13,263700000.0,2012.0,632.0,6.6,2.35,24000
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3848,Color,Shane Carruth,143.0,77.0,291.0,8.0,David Sullivan,291.0,424760.0,Drama|Sci-Fi|Thriller,...,371.0,English,USA,PG-13,7000.0,2004.0,45.0,7.0,1.85,19000
3849,Color,Neill Dela Llana,35.0,80.0,0.0,0.0,Edgar Tancangco,0.0,70071.0,Thriller,...,35.0,English,Philippines,Not Rated,7000.0,2005.0,0.0,6.3,NaN,74
3850,Color,Robert Rodriguez,56.0,81.0,0.0,6.0,Peter Marquardt,121.0,2040920.0,Action|Crime|Drama|Romance|Thriller,...,130.0,Spanish,USA,R,7000.0,1992.0,20.0,6.9,1.37,0
3851,Color,Edward Burns,14.0,95.0,0.0,133.0,Caitlin FitzGerald,296.0,4584.0,Comedy|Drama,...,14.0,English,USA,Not Rated,9000.0,2011.0,205.0,6.4,NaN,413


**Subtask 1.2: Inspect the dataframe**

Inspect the dataframe's columns, shapes, variable types etc.

In [143]:
# Write your code for inspection here
movies.head()

,color,director_name,num_critic_for_reviews,duration,director_facebook_likes,actor_3_facebook_likes,actor_2_name,actor_1_facebook_likes,gross,genres,...,num_user_for_reviews,language,country,content_rating,budget,title_year,actor_2_facebook_likes,imdb_score,aspect_ratio,movie_facebook_likes
0,Color,James Cameron,723.0,178.0,0.0,855.0,Joel David Moore,1000.0,760505847.0,Action|Adventure|Fantasy|Sci-Fi,...,3054.0,English,USA,PG-13,237000000.0,2009.0,936.0,7.9,1.78,33000
1,Color,Gore Verbinski,302.0,169.0,563.0,1000.0,Orlando Bloom,40000.0,309404152.0,Action|Adventure|Fantasy,...,1238.0,English,USA,PG-13,300000000.0,2007.0,5000.0,7.1,2.35,0
2,Color,Sam Mendes,602.0,148.0,0.0,161.0,Rory Kinnear,11000.0,200074175.0,Action|Adventure|Thriller,...,994.0,English,UK,PG-13,245000000.0,2015.0,393.0,6.8,2.35,85000
3,Color,Christopher Nolan,813.0,164.0,22000.0,23000.0,Christian Bale,27000.0,448130642.0,Action|Thriller,...,2701.0,English,USA,PG-13,250000000.0,2012.0,23000.0,8.5,2.35,164000
4,Color,Andrew Stanton,462.0,132.0,475.0,530.0,Samantha Morton,640.0,73058679.0,Action|Adventure|Sci-Fi,...,738.0,English,USA,PG-13,263700000.0,2012.0,632.0,6.6,2.35,24000


In [144]:
movies.tail()

,color,director_name,num_critic_for_reviews,duration,director_facebook_likes,actor_3_facebook_likes,actor_2_name,actor_1_facebook_likes,gross,genres,...,num_user_for_reviews,language,country,content_rating,budget,title_year,actor_2_facebook_likes,imdb_score,aspect_ratio,movie_facebook_likes
3848,Color,Shane Carruth,143.0,77.0,291.0,8.0,David Sullivan,291.0,424760.0,Drama|Sci-Fi|Thriller,...,371.0,English,USA,PG-13,7000.0,2004.0,45.0,7.0,1.85,19000
3849,Color,Neill Dela Llana,35.0,80.0,0.0,0.0,Edgar Tancangco,0.0,70071.0,Thriller,...,35.0,English,Philippines,Not Rated,7000.0,2005.0,0.0,6.3,NaN,74
3850,Color,Robert Rodriguez,56.0,81.0,0.0,6.0,Peter Marquardt,121.0,2040920.0,Action|Crime|Drama|Romance|Thriller,...,130.0,Spanish,USA,R,7000.0,1992.0,20.0,6.9,1.37,0
3851,Color,Edward Burns,14.0,95.0,0.0,133.0,Caitlin FitzGerald,296.0,4584.0,Comedy|Drama,...,14.0,English,USA,Not Rated,9000.0,2011.0,205.0,6.4,NaN,413
3852,Color,Jon Gunn,43.0,90.0,16.0,16.0,Brian Herzlinger,86.0,85222.0,Documentary,...,84.0,English,USA,PG,1100.0,2004.0,23.0,6.6,1.85,456


In [145]:
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3853 entries, 0 to 3852
Data columns (total 28 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   color                      3851 non-null   object 
 1   director_name              3853 non-null   object 
 2   num_critic_for_reviews     3852 non-null   float64
 3   duration                   3852 non-null   float64
 4   director_facebook_likes    3853 non-null   float64
 5   actor_3_facebook_likes     3847 non-null   float64
 6   actor_2_name               3852 non-null   object 
 7   actor_1_facebook_likes     3853 non-null   float64
 8   gross                      3853 non-null   float64
 9   genres                     3853 non-null   object 
 10  actor_1_name               3853 non-null   object 
 11  movie_title                3853 non-null   object 
 12  num_voted_users            3853 non-null   int64  
 13  cast_total_facebook_likes  3853 non-null   int64

In [146]:
movies.describe()

,num_critic_for_reviews,duration,director_facebook_likes,actor_3_facebook_likes,actor_1_facebook_likes,gross,num_voted_users,cast_total_facebook_likes,facenumber_in_poster,num_user_for_reviews,budget,title_year,actor_2_facebook_likes,imdb_score,aspect_ratio,movie_facebook_likes
count,3852.000000,3852.000000,3853.000000,3847.000000,3853.000000,3.853000e+03,3.853000e+03,3853.000000,3847.000000,3853.000000,3.853000e+03,3853.000000,3852.000000,3853.000000,3781.000000,3853.000000
mean,163.046989,109.921080,784.507397,747.290096,7575.558526,5.096547e+07,1.024240e+05,11239.480145,1.378217,326.720478,4.524817e+07,2003.061511,1959.018432,6.462886,2.109569,9089.546068
std,123.927258,22.741728,3027.395958,1841.623945,15405.383114,6.932033e+07,1.502942e+05,18922.948757,2.056071,407.920904,2.234208e+08,10.007168,4472.171290,1.053843,0.353236,21277.375223
min,1.000000,34.000000,0.000000,0.000000,0.000000,1.620000e+02,5.000000e+00,0.000000,0.000000,1.000000,2.180000e+02,1920.000000,0.000000,1.600000,1.180000,0.000000
25%,72.000000,95.000000,10.000000,183.000000,721.000000,6.830957e+06,1.730900e+04,1817.000000,0.000000,102.000000,1.000000e+07,1999.000000,362.000000,5.900000,1.850000,0.000000
50%,134.000000,106.000000,58.000000,427.000000,1000.000000,2.790000e+07,5.052300e+04,3876.000000,1.000000,203.000000,2.400000e+07,2005.000000,664.000000,6.600000,2.350000,206.000000
75%,221.000000,120.000000,222.000000,685.000000,12000.000000,6.550000e+07,1.241850e+05,15972.000000,2.000000,391.000000,5.000000e+07,2010.000000,971.000000,7.200000,2.350000,11000.000000
max,813.000000,330.000000,23000.000000,23000.000000,640000.000000,7.605058e+08,1.689764e+06,656730.000000,43.000000,5060.000000,1.221550e+10,2016.000000,137000.000000,9.300000,16.000000,349000.000000


#### <font color='red'>Question 1: How many rows and columns are present in the dataframe? </font>
-  <font color='red'>(3821, 26)</font>
-  <font color='red'>(3879, 28)</font>
-  <font color='red'>(3853, 28)</font>
-  <font color='red'>(3866, 26)</font>

In [147]:
## Write your code here
movies.shape


(3853, 28)

#### <font color='red'>Question 2: How many columns have null values present in them? Try writing a code for this instead of counting them manually.</font>

-  <font color='red'>3</font>
-  <font color='red'>6</font>
-  <font color='red'>9</font>
-  <font color='red'>12</font>

In [148]:
## Write your code here
null_values = movies.isnull().sum()
len(null_values[null_values>0])




12

### Task 2: Cleaning the Data

**Subtask 2.1: Drop unecessary columns**

For this assignment, you will mostly be analyzing the movies with respect to the ratings, gross collection, popularity of movies, etc. So many of the columns in this dataframe are not required. So it is advised to drop the following columns.
-  color
-  director_facebook_likes
-  actor_1_facebook_likes
-  actor_2_facebook_likes
-  actor_3_facebook_likes
-  actor_2_name
-  cast_total_facebook_likes
-  actor_3_name
-  duration
-  facenumber_in_poster
-  content_rating
-  country
-  movie_imdb_link
-  aspect_ratio
-  plot_keywords

#### <font color='red'>Question 3: What is the count of columns in the new dataframe? </font>
-  <font color='red'>10</font>
-  <font color='red'>13</font>
-  <font color='red'>15</font>
-  <font color='red'>17</font>

In [149]:
## Write your code here
columns_to_drop = ["color","director_facebook_likes","actor_1_facebook_likes",
"actor_2_facebook_likes","actor_3_facebook_likes","actor_2_name","cast_total_facebook_likes"
,"actor_3_name","duration","facenumber_in_poster","content_rating","country","movie_imdb_link"
,"aspect_ratio","plot_keywords"]


In [150]:
movies.drop(columns=columns_to_drop, axis=1, inplace= True)

In [151]:
movies.columns

Index(['director_name', 'num_critic_for_reviews', 'gross', 'genres',
       'actor_1_name', 'movie_title', 'num_voted_users',
       'num_user_for_reviews', 'language', 'budget', 'title_year',
       'imdb_score', 'movie_facebook_likes'],
      dtype='object')

**Subtask 2.2: Inspect Null values**

As you have seen above, there are null values in multiple columns of the dataframe 'movies'. Find out the percentage of null values in each column of the dataframe 'movies'.

In [152]:
## Write your code here
missing_percentage=round((movies.isnull().sum()/len(movies))*100,2)
missing_percentage



,0
director_name,0.00
num_critic_for_reviews,0.03
gross,0.00
genres,0.00
actor_1_name,0.00
movie_title,0.00
num_voted_users,0.00
num_user_for_reviews,0.00
language,0.10
budget,0.00


#### <font color='red'>Question 4: Which column has the highest percentage of null values? </font>
-  <font color='red'>language</font>
-  <font color='red'>genres</font>
-  <font color='red'>num_critic_for_reviews</font>
-  <font color='red'>imdb_score</font>

In [153]:
## Write your code here
missing_percentage.idxmax()

'language'

**Subtask 2.3: Fill NaN values**

You might notice that the `language` column has some NaN values. Here, on inspection, you will see that it is safe to replace all the missing values with `'English'`.

In [154]:
# Write your code for filling the NaN values in the 'language' column here
movies["language"]=movies["language"].fillna("English")
#Checking the missing values after filling
movies["language"].isnull().sum()


np.int64(0)

#### <font color='red'>Question 5: What is the count of movies made in English language after replacing the NaN values with English? </font>
-  <font color='red'>3670</font>
-  <font color='red'>3674</font>
-  <font color='red'>3668</font>
-  <font color='red'>3672</font>

In [155]:
## Write your code here
english_movies= movies[movies["language"]=="English"]
len(english_movies["language"])


3675

### Task 3: Data Analysis

**Subtask 3.1: Change the unit of columns**

Convert the unit of the `budget` and `gross` columns from `$` to `million $`.

In [156]:
## Write your code here
movies[['budget', 'gross']]=movies[['budget', 'gross']]/1000000

In [157]:
#Check the conversion
movies[['budget', 'gross']]

,budget,gross
0,237.0000,760.505847
1,300.0000,309.404152
2,245.0000,200.074175
3,250.0000,448.130642
4,263.7000,73.058679
...,...,...
3848,0.0070,0.424760
3849,0.0070,0.070071
3850,0.0070,2.040920
3851,0.0090,0.004584


**Subtask 3.2: Find the movies with highest profit**

   1. Create a new column called `profit` which contains the difference of the two columns: `gross` and `budget`.
   2. Sort the dataframe using the `profit` column as reference. (Find which command can be used here to sort entries from the documentation)
   3. Extract the top ten profiting movies in descending order and store them in a new dataframe - `top10`

In [158]:
# Write your code for creating the profit column here
movies["profit"]=movies["gross"]-movies["budget"]
#Check the change
movies[["profit","gross","budget"]]


,profit,gross,budget
0,523.505847,760.505847,237.0000
1,9.404152,309.404152,300.0000
2,-44.925825,200.074175,245.0000
3,198.130642,448.130642,250.0000
4,-190.641321,73.058679,263.7000
...,...,...,...
3848,0.417760,0.424760,0.0070
3849,0.063071,0.070071,0.0070
3850,2.033920,2.040920,0.0070
3851,-0.004416,0.004584,0.0090


In [159]:
#top10 = # Write your code to get the top 10 profiting movies here
#help(pd.DataFrame.sort_values)
movies.sort_values(by='profit', ascending = False, inplace=True)


**Checkpoint:** You might spot two movies directed by `James Cameron` in the list.

#### <font color='red'>Question 6: Which movie is ranked 5th from the top in the list obtained? </font>
-  <font color='red'>E.T. the Extra-Terrestrial</font>
-  <font color='red'>The Avengers</font>
-  <font color='red'>The Dark Knight</font>
-  <font color='red'>Titanic</font>

In [160]:
## Write your code here
movies.iloc[4]["movie_title"]

'E.T. the Extra-Terrestrial\xa0'

**Subtask 3.3: Find IMDb Top 200**

Create a new dataframe `IMDb_Top_200` and store the top 200 movies with the highest IMDb Rating (corresponding to the column: `imdb_score`). Also make sure that for all of these movies, the `num_voted_users` is greater than 25,000.

Also add a `Rank` column containing the values 1 to 200 indicating the ranks of the corresponding films.

In [161]:
# Write your code for extracting the top 200 movies as per the IMDb score here. Make sure that you store it in a new dataframe
# and name that dataframe as 'IMDb_Top_200'
IMDB_Top_200 = movies[movies["num_voted_users"] > 25000].sort_values(
 by="imdb_score", ascending=False).head(200)

IMDB_Top_200["Rank"] = range(1, 201)




In [162]:
#Check if all num_voted_users is greater than 25000
(IMDB_Top_200["num_voted_users"]>25000).all()

np.True_

#### <font color='red'>Question 7: Suppose movies are divided into 5 buckets based on the IMDb ratings: </font>
-  <font color='red'>7.5 to 8</font>
-  <font color='red'>8 to 8.5</font>
-  <font color='red'>8.5 to 9</font>
-  <font color='red'>9 to 9.5</font>
-  <font color='red'>9.5 to 10</font>

<font color = 'red'> Which bucket holds the maximum number of movies from *IMDb_Top_200*? </font>

In [163]:
## Write your code here
#help(pd.cut)
bins = [7.5, 8, 8.5, 9, 9.5, 10]
labels = ['7.5 to 8', '8 to 8.5', '8.5 to 9', '9 to 9.5', '9.5 to 10']

IMDB_Top_200["score_bucket"] = pd.cut(IMDB_Top_200["imdb_score"],
                                      bins=bins,
                                      labels=labels,
                                      include_lowest=True)

bucket_counts = IMDB_Top_200["score_bucket"].value_counts().sort_index()
bucket_counts

,count
score_bucket,
7.5 to 8,48
8 to 8.5,124
8.5 to 9,26
9 to 9.5,2
9.5 to 10,0


In [164]:
bucket_counts.idxmax()

'8 to 8.5'

**Subtask 3.4: Find the critic-favorite and audience-favorite actors**

   1. Create three new dataframes namely, `Meryl_Streep`, `Leo_Caprio`, and `Brad_Pitt` which contain the movies in which the actors: 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' are the lead actors. Use only the `actor_1_name` column for extraction. Also, make sure that you use the names 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' for the said extraction.
   2. Append the rows of all these dataframes and store them in a new dataframe named `Combined`.
   3. Group the combined dataframe using the `actor_1_name` column.
   4. Find the mean of the `num_critic_for_reviews` and `num_user_for_review` and identify the actors which have the highest mean.

In [165]:
# Write your code for creating three new dataframes here
#Meryl_Streep = # Include all movies in which Meryl_Streep is the lead
Merly_Streep =movies[movies["actor_1_name"]=="Meryl Streep"]
Merly_Streep.head(5)


,director_name,num_critic_for_reviews,gross,genres,actor_1_name,movie_title,num_voted_users,num_user_for_reviews,language,budget,title_year,imdb_score,movie_facebook_likes,profit
1322,David Frankel,208.0,124.732962,Comedy|Drama|Romance,Meryl Streep,The Devil Wears Prada,286178,631.0,English,35.0,2006.0,6.8,0,89.732962
1471,Sydney Pollack,66.0,87.100000,Biography|Drama|Romance,Meryl Streep,Out of Africa,52339,200.0,English,31.0,1985.0,7.2,0,56.100000
1132,Nora Ephron,252.0,94.125426,Biography|Drama|Romance,Meryl Streep,Julie & Julia,79264,277.0,English,40.0,2009.0,7.0,13000,54.125426
1514,David Frankel,234.0,63.536011,Comedy|Drama|Romance,Meryl Streep,Hope Springs,34258,178.0,English,30.0,2012.0,6.3,0,33.536011
392,Nancy Meyers,187.0,112.703470,Comedy|Drama|Romance,Meryl Streep,It's Complicated,69860,214.0,English,85.0,2009.0,6.6,0,27.703470


In [166]:
#Leo_Caprio = # Include all movies in which Leo_Caprio is the lead
Leo_Caprio = movies[movies["actor_1_name"]=="Leonardo DiCaprio"]
Leo_Caprio.head(5)


,director_name,num_critic_for_reviews,gross,genres,actor_1_name,movie_title,num_voted_users,num_user_for_reviews,language,budget,title_year,imdb_score,movie_facebook_likes,profit
25,James Cameron,315.0,658.672302,Drama|Romance,Leonardo DiCaprio,Titanic,793059,2528.0,English,200.0,1997.0,7.7,26000,458.672302
94,Christopher Nolan,642.0,292.568851,Action|Adventure|Sci-Fi|Thriller,Leonardo DiCaprio,Inception,1468200,2803.0,English,160.0,2010.0,8.8,175000,132.568851
859,Steven Spielberg,194.0,164.435221,Biography|Crime|Drama,Leonardo DiCaprio,Catch Me If You Can,525801,667.0,English,52.0,2002.0,8.0,15000,112.435221
283,Quentin Tarantino,765.0,162.804648,Drama|Western,Leonardo DiCaprio,Django Unchained,955174,1193.0,English,100.0,2012.0,8.5,199000,62.804648
173,Alejandro G. Iñárritu,556.0,183.635922,Adventure|Drama|Thriller|Western,Leonardo DiCaprio,The Revenant,406020,1188.0,English,135.0,2015.0,8.1,190000,48.635922


In [167]:
#Brad_Pitt = # Include all movies in which Brad_Pitt is the lead
Brad_Pitt = movies[movies["actor_1_name"]=="Brad Pitt"]
Brad_Pitt.head(5)


,director_name,num_critic_for_reviews,gross,genres,actor_1_name,movie_title,num_voted_users,num_user_for_reviews,language,budget,title_year,imdb_score,movie_facebook_likes,profit
383,Steven Soderbergh,186.0,183.405771,Crime|Thriller,Brad Pitt,Ocean's Eleven,402645,845.0,English,85.0,2001.0,7.8,0,98.405771
244,Doug Liman,233.0,186.336103,Action|Comedy|Crime|Romance|Thriller,Brad Pitt,Mr. & Mrs. Smith,348861,798.0,English,120.0,2005.0,6.5,0,66.336103
887,Neil Jordan,120.0,105.264608,Drama|Fantasy|Horror,Brad Pitt,Interview with the Vampire: The Vampire Chroni...,239752,406.0,English,60.0,1994.0,7.6,11000,45.264608
448,David Ayer,406.0,85.707116,Action|Drama|War,Brad Pitt,Fury,303185,701.0,English,68.0,2014.0,7.6,82000,17.707116
243,Steven Soderbergh,198.0,125.531634,Crime|Thriller,Brad Pitt,Ocean's Twelve,284852,627.0,English,110.0,2004.0,6.4,0,15.531634


In [168]:
# Write your code for combining the three dataframes here
Combined = pd.concat([Merly_Streep,Leo_Caprio,Brad_Pitt], axis=0)




In [169]:
# Write your code for grouping the combined dataframe here
grouped_df= Combined.groupby("actor_1_name")


In [170]:
# Write the code for finding the mean of critic reviews and audience reviews here
mean_reviews = grouped_df.mean(["num_critic_for_reviews", "num_user_for_reviews"])
mean_reviews



,num_critic_for_reviews,gross,num_voted_users,num_user_for_reviews,budget,title_year,imdb_score,movie_facebook_likes,profit
actor_1_name,,,,,,,,,
Brad Pitt,245.000000,66.321449,283583.823529,742.352941,69.294118,2004.352941,7.152941,14051.764706,-2.972669
Leonardo DiCaprio,330.190476,125.741989,441958.857143,914.476190,82.452381,2005.476190,7.495238,52190.476190,43.289608
Meryl Streep,181.454545,59.919727,73545.545455,297.181818,34.454545,2003.545455,6.745455,2934.090909,25.465182


#### <font color='red'>Question 8: Which actor is highest rated among the three actors according to the user reviews? </font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

In [171]:
## Write your code here
critic_favorite = mean_reviews["num_user_for_reviews"].idxmax()
critic_favorite


'Leonardo DiCaprio'

#### <font color='red'>Question 9: Which actor is highest rated among the three actors according to the critics?</font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

In [172]:
## Write your code here
critic_favorite = mean_reviews["num_critic_for_reviews"].idxmax()
critic_favorite

'Leonardo DiCaprio'

#### <font color='red'>Question 10: Which movie starting with the letter A, has the 3rd highest profit?</font>
-  <font color='red'>Austin Powers: The Spy Who Shagged Me</font>
-  <font color='red'>Avatar</font>
-  <font color='red'>Avengers: Age of Ultron</font>

In [173]:
## Write your code here
A_movies = movies[movies["movie_title"].str.startswith("A")]
A_movies.sort_values("profit", ascending=False).iloc[2]["movie_title"]


'Avengers: Age of Ultron\xa0'

#### <font color='red'>Question 11: Among the movies starting with the letter A, who has the highest imdb score?</font>
-  <font color='red'>American History X</font>
-  <font color='red'>Apocalypse Now</font>
-  <font color='red'>Avengers: Age of Ultron</font>

In [174]:
## Write your code here
A_movies.sort_values("imdb_score", ascending=False).iloc[0]["movie_title"]


'American History X\xa0'

#### <font color='red'>Question 12: Among the movies that '*Joss Whedon*' has directed, which of them made the highest profit?</font>
-  <font color='red'>Serenity</font>
-  <font color='red'>The Avengers</font>
-  <font color='red'>Avengers: Age of Ultron</font>

In [175]:
## Write your code here
Joss_movies = movies[movies["director_name"]=="Joss Whedon"]
Joss_movies.sort_values("profit", ascending=False).iloc[0]["movie_title"]

'The Avengers\xa0'

#### <font color='red'>Question 13: Among the movies that were released in the year '*2007*' & '*2008*', which of them had the 2nd highest imdb score?</font>
-  <font color='red'>The Dark Knight</font>
-  <font color='red'>WALL·E</font>
-  <font color='red'>No End in Sight</font>
-  <font color='red'>Into the Wild</font>

In [176]:
## Write your code here
movies_0708 = movies[(movies["title_year"]==2007) | (movies["title_year"]==2008)]
movies_0708.sort_values("imdb_score", ascending=False).iloc[1]["movie_title"]


'WALL·E\xa0'